packages/amplify-category-interactions/src/provider-utils/awscloudformation/service-walkthroughs/lex-walkthrough.js

```diff
@@ -544,45 +419,54 @@ async function addSlot(context, intentName, botName, resourceName, serviceMetada
       slot.type = slot.type[0];
     }
 
-    slot.prompt = await inquirer.prompt(slotPromptQuestion);
-    slot.prompt = slot.prompt[inputs[16].key];
-
-    slot.required = await inquirer.prompt(slotRequiredQuestion);
-    slot.required = slot.required[inputs[17].key];
+    slot.prompt = await prompter.prompt(
+      'Enter a prompt for your slot (e.g. What city?)',
+      {
+        initial: resourceName,
+        validate: matchRegex(/^.{1,1000}$/, 'Prompts can have a maximum of 1000 characters and cannot be empty'),
+      }
+    );
+    slot.required = await prompter.yesOrNo('Should this slot be required?', true);
 
     slots.push(slot);
+  } while (await prompter.yesOrNo('Would you like to add another slot?', true));
 
-    addAnotherSlot = await inquirer.prompt(addAnotherSlotQuestion);
-    addAnotherSlot = addAnotherSlot[inputs[25].key];
-  }
   if (newSlotTypeAdded) {
     return [slots, newSlotTypes];
   }
+
   return [slots];
 }
 
-async function getSlotType(context, serviceMetadata, newSlotTypes, parameters) {
-  const { inputs } = serviceMetadata;
-  const { amplify, print } = context;
+async function askSlotName(resourceName) {
+  return prompter.input(
+    'Enter a name for your slot (e.g. Location)',
+    {
+      initial: resourceName,
+      validate: matchRegex(
+        /^([A-Za-z]_?){1,100}$/,
+        'Slot name can only contain letters, must be no longer than 100 characters, and cannot be empty'
+      ),
+    }
+  );
+}
+
+async function getSlotType(context, newSlotTypes, parameters) {
   let slotType;
-  inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'));
 
-  const slotTypeChoiceQuestion = {
-    type: inputs[26].type,
-    name: inputs[26].key,
-    message: inputs[26].question,
-    choices: inputs[26].options,
-  };
-  const slotTypeChoice = await inquirer.prompt(slotTypeChoiceQuestion);
+  const slotTypeChoice = await prompter.pick(
+    'Would you like to choose an Amazon built-in slot type, a slot type you\'ve already made, or create a new slot type?',
+    ['Amazon built-in slot type', "Slot type I've already made", 'Create a new slot type']
+  );
   function searchSlotTypes(builtInSlotTypes) {
```

Comment:
CodeQL says this function is not used anywhere? If so, remove it

Reply:
removed